### PR TITLE
Tests : timedata : median file filter shall not grow beyond size

### PR DESCRIPTION
**Pull request by mmachicao**
Test the contract of CMedianFilter and it's behavior when adding new entries into the filter.

- List of entries has fixed size
- New entries are added at the end.
- Surplus entries are removed from the front.

### CODE REVIEW

File: src/test/timedata_tests.cpp [hunk 1]
```diff
@@ -34,4 +34,22 @@ BOOST_AUTO_TEST_CASE(util_MedianFilter)
     BOOST_CHECK_EQUAL(filter.median(), 7);
 }
 
+
+BOOST_AUTO_TEST_CASE(util_MedianFilterShallNotGrowBeyondSize)
+{
+
+  CMedianFilter<int> filter(2,15);
```
Comment (practicalswift) [inline src/test/timedata_tests.cpp]: Run this through `clang-format` on the new code to make sure the formatting is consistent with the style guide :-)

Comment (laanwj) [reply src/test/timedata_tests.cpp]: Also: please use four spaces for indentation

**Comment by laanwj** [comment]
utACK, apart from style nits

File: src/test/timedata_tests.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/test/timedata_tests.cpp]: See https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy

**Comment by MarcoFalke** [comment]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Comment by mmachicao** [comment]
Sorry guys. Got confused with git. Closing this request and replacing with a new one matching your requirements on a different branch. At this moment I am more interested on contributing the test rather than dealing with git.

**Pull request closed without merging**